unified-prototype/unified-plugin/plugin-plugin/build.gradle.kts

```
@@ -0,0 +1,51 @@
+@file:Suppress("UnstableApiUsage")
+
+plugins {
+    `kotlin-dsl`
+    id("build-logic.publishing")
+    groovy // For spock testing
+}
+
+description = "Implements the declarative Plugin building Plugin"
+
+dependencies {
+    implementation(project(":plugin-common"))
+    implementation("org.gradle.toolchains:foojay-resolver:0.8.0")
+    implementation(gradleApi())
+}
+
+testing {
+    suites {
+        @Suppress("UnstableApiUsage")
+        val integTest by registering(JvmTestSuite::class) {
+            useSpock("2.2-groovy-3.0")
+
+            dependencies {
+                implementation(project(":internal-testing-utils"))
+            }
+        }
+
+        tasks.getByPath("check").dependsOn(integTest)
+    }
+}
+
+gradlePlugin {
+    testSourceSets(project.sourceSets.getByName("integTest"))
+
+    plugins {
+        create("gradle-plugin") {
+            id = "org.gradle.experimental.gradle-plugin"
```

Just a suggestion... 

name tweaks...
```suggestion
            id = "org.gradle.experimental.java-gradle-plugin"
```